### PR TITLE
Specify build-system dependencies

### LINKED ISSUE #49: Installation from source fails
Installing with pip option `no-binary` results on error.

## How to reproduce
In a virtualenv without setuptools-git-versioning installed, run `pip install --no-binary :all: setuptools-git-versioning`

The error will be something like this

```
  ERROR: Command errored out with exit status 1:
   command: /home/bciconel/SourceCode/setuptools-git-versioning/venv/bin/python /home/bciconel/SourceCode/setuptools-git-versioning/venv/lib64/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpaszqbvaf
       cwd: /tmp/pip-req-build-4qnro2f3
  Complete output (20 lines):
  Traceback (most recent call last):
    File "/home/bciconel/SourceCode/setuptools-git-versioning/venv/lib64/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 349, in <module>
      main()
    File "/home/bciconel/SourceCode/setuptools-git-versioning/venv/lib64/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 331, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/home/bciconel/SourceCode/setuptools-git-versioning/venv/lib64/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 117, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-suhsdn2l/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 177, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "/tmp/pip-build-env-suhsdn2l/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 159, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-suhsdn2l/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 281, in run_setup
      super(_BuildMetaLegacyBackend,
    File "/tmp/pip-build-env-suhsdn2l/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 174, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 3, in <module>
      from setuptools_git_versioning import version_from_git
    File "/tmp/pip-req-build-4qnro2f3/setuptools_git_versioning.py", line 13, in <module>
      import toml
  ModuleNotFoundError: No module named 'toml'
  ----------------------------------------
  ```

### PR DESCRIPTION
**Pull request by bruno-fs**
Also, only import from internal module on setup.py after adding its path
to pythonpath.

Closes #49.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,13 @@
+[build-system]
```
Comment (dolfinus) [inline pyproject.toml:1]: Please undo this change

File: setup.py [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+from setuptools import find_packages, setup
 
 HERE = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(HERE)
+
+
+def infer_version():
+    from setuptools_git_versioning import version_from_git
+
+    return version_from_git()
```
Comment (dolfinus) [inline setup.py:13]:
```suggestion
from setuptools_git_versioning import version_from_git
```

File: setup.py [hunk 3]
```diff
@@ -18,7 +26,7 @@ def parse_requirements(file_content):
 
 setup(
     name="setuptools-git-versioning",
-    version=version_from_git(),
+    version=infer_version(),
```
Comment (dolfinus) [inline setup.py:29]:
```suggestion
    version=version_from_git(),
```

File: pyproject.toml (hunk 1 above)
Comment (bruno-fs) [reply pyproject.toml:1]: I agree, but setting up build-system was the only way I found to prevent the error described on #49

Comment (bruno-fs) [reply pyproject.toml:1]: When building from source these depends need to be installed on the isolated build environment. It is impossible to execute `setup.py` since the dependencies will only be installed after importing the function `version_from_git`

Comment (bruno-fs) [reply pyproject.toml:1]: I opened another PR that should reproduce this problem: #51

File: setup.py [hunk 4]
```diff
@@ -18,7 +23,7 @@ def parse_requirements(file_content):
 
 setup(
     name="setuptools-git-versioning",
-    version=version_from_git(),
```
Comment (dolfinus) [inline setup.py:21]: Please undo

File: tests/lib/util.py [hunk 5]
```diff
@@ -147,7 +147,7 @@ def create_pyproject_toml(
             """
         ),
         commit=False,
-        **kwargs
+        **kwargs,
```
Comment (dolfinus) [inline tests/lib/util.py:150]: Please undo

File: tests/lib/util.py [hunk 6]
```diff
@@ -215,7 +215,7 @@ def create_setup_py(
                 coverage.save()
             """
         ).format(cfg=cfg),
-        **kwargs
+        **kwargs,
```
Comment (dolfinus) [inline tests/lib/util.py:218]: Please undo

File: pyproject.toml [hunk 7]
```diff
@@ -1,10 +1,20 @@
...
+build-backend = "setuptools.build_meta"
+
 [tool.isort]
 profile = "black"
 multi_line_output = 3
 
 [tool.black]
 line-length = 120
-target-version = ['py27', 'py35', 'py36', 'py37', 'py38', 'py39', 'py310']
+target-version = ['py35', 'py36', 'py37', 'py38', 'py39', 'py310']
```
Comment (dolfinus) [inline pyproject.toml:17]: Please undo

File: tests/lib/util.py (hunk 5 above)
Comment (bruno-fs) [reply tests/lib/util.py:150]: mmm I didn't author the changes on this file. This was done via pre-commit ci :thinking: https://github.com/dolfinus/setuptools-git-versioning/pull/50/commits/8d90b979554821e3b670d7300a58ee370fb81efe

File: setup.py (hunk 4 above)
Comment (bruno-fs) [reply setup.py:21]: ouch :sweat_smile:

File: pyproject.toml (hunk 1 above)
Comment (bruno-fs) [reply pyproject.toml:1]: Also note I didn't include all dependencies. Only the ones required for setup.
This is equivalent to `setup_requires` on `setup.py` (but this wouldn't solve the problem because setup.py would break before setting build dependencies :shrug:

File: tests/lib/util.py (hunk 5 above)
Comment (bruno-fs) [reply tests/lib/util.py:150]: Do you mind if I just drop this commit and force push?
